src/main/java/com/limechain/rpc/subscriptions/author/AuthorRpcImpl.java

```diff
@@ -35,7 +41,19 @@ public AuthorRpcImpl(String forwardNodeAddress, AuthorRPC authorRPC) {
 
     @Override
     public void authorSubmitAndWatchExtrinsic(String extrinsic) {
-        authorRPC.authorSubmitExtrinsic(extrinsic);
+        Extrinsic decodedExtrinsic = new Extrinsic(
+                ScaleUtils.Decode.decode(
+                        StringUtils.hexToBytes(extrinsic),
+                        ScaleCodecReader::readByteArray
+                )
+        );
+
+        try {
+            transactionProcessor.handlePoolOnlyExternalTransaction(decodedExtrinsic);
+        } catch (TransactionValidationException e) {
+            throw new ExecutionFailedException("Failed to executed submit_extrinsic call: " + e.getMessage());
+        }
+
         rpcClient.send(SubscriptionName.AUTHOR_SUBMIT_AND_WATCH_EXTRINSIC.getValue(),
```

Comment:
It seems that our implementation doesn't cover past the "submit" part of the RPC call. Currently we receive an extrinsic and create a WS subscription on the `author_extrinsicUpdate` topic. Afterwards no following updates are sent. As [per spec](https://github.com/w3f/PSPs/blob/master/PSPs/drafts/psp-6.md#189-author_submitandwatchextrinsic-pubsub)  we should notify the user on the different states of the extrinsic lifetime. See the usage of `Topic.CHAIN_ALL_HEAD` for example.

Reply:
You're correct, but currently, we are not supporting all of the states provided in the provided link, so I decided to return the responses from the smaldot. So if you don't mind, I would suggest sticking with the current solution and changing it at a later stage.